src/socket/unix_server.cr

```diff
@@ -43,8 +45,8 @@ class UNIXServer < UNIXSocket
   # server socket when the block returns.
   #
   # Returns the value of the block.
-  def self.open(path, type : Type = Type::STREAM, backlog = 128)
-    server = new(path, type, backlog)
+  def self.open(path, type : Type = Type::STREAM, backlog = 128, abstract is_abstract = false)
```

Comment:
For this to work well we would need to drop the Lexer/Parser mothod and use something like a Parsing Expression Grammar (PEG) parser...